test/combinations.jl

```diff
@@ -1,429 +1,365 @@
-# This section tests for:
-# - All supported combinations of integral(f, ::Geometry, ::IntegrationAlgorithm) produce accurate results
-# - Invalid applications of integral aliases (e.g. lineintegral) produce a descriptive error
+"""
+This file includes tests for:
+- Supported combinations {::Geometry, ::IntegrationRule} for `integral(f, geometry, rule)`.
+- Integrand functions returning a `Unitful.Quantity` or a `Vector{Unitful.Quantity}`.
+- Callable objects in place of an integrand function.
+- Unsupported combinations produce a descriptive and useful error message.
+- The appropriate alias function, e.g. `lineintegral` for 1D geometries, works.
+- Invalid applications of integral aliases produce a descriptive and useful error message.
+- (Planned) Usage of non-default DifferentiationMethods.
+"""
+
+#===============================================================================
+                        Test Generation Infrastructure
+===============================================================================#
+
+@testsnippet Combinations begin
```

Comment:
Does this mean, we can delete the `Setup` `@testsnippet` in runtests.jl?

Reply:
The `@testsnippet Setup` was still used in `test/utils.jl` and `test/floating_point_types.jl`, but after seeing this comment and thinking about it some more, I decided to re-organize a bit. Now we have `@testsnippet`s in each test file where they're required so that all relevant code is in the same source file.

Reply:
Sounds good.